src/client/network/raw_socket.ts

```diff
@@ -0,0 +1,41 @@
+import * as io from 'socket.io-client'
+import Socket = SocketIOClient.Socket
+import { injectable } from 'inversify'
+
+@injectable()
+export class RawSocket {
+  private socket: Socket
+
+  public static of() {
+    return new RawSocket()
+  }
+
+  public on(event: string, fn: (...args: any[]) => void) {
+    this.connectIfNotConnected()
```

Comment:
I avoid constructors doing any kind of work. It makes classes much easier to test because the constructor doesn't attempt to perform any side effects, such as networking.

I could add an explicit `connect()` method, but since I'm making everything else in networking lazy, I think it also makes sense to only connect if someone actually tries to listen for something on the network as well.